tests/unit/plotting/traceplot_test.py

```diff
@@ -204,24 +204,49 @@ def test_traceplot_no_experiment(gaze_no_exp):
     pm.plotting.traceplot(gaze_no_exp, show=False)
 
 
+def test_set_screen_axes_valid(gaze):
+    fig, ax = pm.plotting.traceplot(
+        gaze=gaze,
+        show=False,
+    )
+    assert ax.get_xlim() == (0, gaze.experiment.screen.width_px)
+    assert ax.get_ylim() == (gaze.experiment.screen.height_px, 0)
+    assert ax.get_aspect() == 1
+    plt.close(fig)
+
+
+@pytest.mark.parametrize('origin', ['lower left', 'center', 'upper right'])
+def test_set_screen_axes_invalid_origin(origin, gaze):
+    gaze.experiment.screen.origin = origin
+    with pytest.raises(ValueError, match='screen origin must be "upper left"'):
+        pm.plotting.traceplot(gaze=gaze, show=False)
+
+
 @pytest.mark.parametrize(
     'width,height',
     [
-        (1280, 1024),  # both defined
-        (None, 1024),   # width None
-        (1280, None),   # height None
-        (None, None),   # both None
+        (None, None),
+        (None, 768),
+        (1024, None),
     ],
 )
-def test_traceplot_screen_dims(gaze, width, height):
-    # Set screen dimensions
+def test_set_screen_axes_none_dimensions_returns(width, height, gaze):
+    """Should not raise or override axes when screen dimensions are None."""
     gaze.experiment.screen.width_px = width
     gaze.experiment.screen.height_px = height
 
-    pm.plotting.traceplot(gaze=gaze, show=False)
+    fig, ax = plt.subplots()
 
```

Comment:
```suggestion
    assert ax.get_aspect() != 'equal'
```

As you check the aspect of the axis later on, you should also check its value before calling the function.